src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,21 +136,7 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-
-                // TODO: https://github.com/dotnet/corefx/issues/19605
-                //
-                //   Catching NotSupportedException here because HostName is currently PNSE
-                //   on UapAot. HostName is planned to be restored for .NetCore 2.1
-                //   so once that's done, the try-catch can be removed.
-                string clientDomainRaw;
-                try
-                {
-                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
-                }
-                catch (NotSupportedException)
-                {
-                    clientDomainRaw = "LocalHost";
-                }
+                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
```

Comment:
There is no test to test against the change. The method Initialize() is private, it will get called every time an instance of SmtpClient() get created.

The `clientDomain` field is internal for SmtpClient (`System.Net.Mail`), and it's not in the same assembly as `System.Net.Mail.Tests`. I am not sure how to write test for it.